### ISSUE TITLE
[Regression] Alphabetic keyboard instead of numeric for measurable habits

### ISSUE THREAD

**Issue by holocronweaver** [iSoron/uhabits#1827]
## Pre-submission checklist
- [x] I am submitting a bug report, not a feature request.
- [x] I have searched for similar issues, but did not find any matches.

## Description
In version 2.1.2, when entering a value for a measurable habit, the numeric keypad is displayed by default.

In version 2.1.3 the regular alphabet keyboard is displayed by default instead, requiring manual switch to a numeric layout each time you want to enter the value for a measurable habit.

Downgrading to 2.1.2 resolves the issue (thankfully backups are backwards compatible in this case).

## Steps to reproduce
1. Install 2.1.3.
2. Add a measurable habit.
3. Click on an entry for the measurable habit to enter a value.
4. Text-entry alphabetic keyboard pops up instead of numeric.
5. Downgrade to 2.1.2.
6. Repeat steps 2-4, note numeric keyboard pops up instead of text-entry alphabetic.

## System information
 - Phone: Pixel 4 XL
 - Phone Operating System: Android 13
 - App version: 2.1.3
 - Keyboard: default or SwiftKey, makes no difference

## Screenshots
Can add if it would help.

**Comment by pililac** [comment]
Same here with Version 2.2.1 on a Pixel 8.

**Comment by Saransh719** [comment]
Hi! I’d love to work on this issue. Is anyone already working on it?

**Comment by YakovL** [comment]
@Saransh719 I think it's safe to assume "no" if the issue is still relevant

**Comment by holocronweaver** [comment]
I am now on v2.3.1 and the issue seems to be resolved for me on Android 16 using Gboard but NOT when using Swiftkey and other keyboards. Swiftkey correctly identifies numeric entry in other apps so not sure what this app is doing different.

Tldr this issue stills needs attention for non Google keyboards

**Comment by YakovL** [comment]
Yeah, looks like a matter of `inputType`, not difficult to test and fix for a specific keyboard (although may be tricky to fix consistently for all the keyboards as some of them may use heuristics, plus for all the inputs) if you have your dev env ready.

To fix, one should understand which file(s) to edit ([less than 6 candidates](https://github.com/search?q=repo%3AiSoron%2Fuhabits%20inputType&type=code)) and try modifying the `inputType`. Ironically enough, there's already [at least one](https://github.com/iSoron/uhabits/blob/2b24759d6f65db8e9d7e7ca5538857d0e4828e24/uhabits-android/src/main/java/org/isoron/uhabits/activities/common/dialogs/NumberDialog.kt#L105) fix for Swiftkey for this matter (maybe it should be reused?).

For your convenience, [here are other "educated guesses" by GPT](https://chatgpt.com/share/68f50e1a-5848-8003-8960-8caeda7bbfeb).

**Comment by holocronweaver** [comment]
I tested HeliBoard and Fossify Keyboard and they too work as expected, showing numpad for numeric fields, so it may indeed be a Swiftkey specific issue at this point!

**Comment by Saransh719** [comment]
> I tested HeliBoard and Fossify Keyboard and they too work as expected, showing numpad for numeric fields, so it may indeed be a Swiftkey specific issue at this point!

I think that's because in the numberdialog.kt file which @YakovL mentioned has specifc code to overwrite swift and samsung keyboards only, I will try to change that and see if its because of a glitch with those keyboards or if there was previously a glitch but now it works.

**Comment by Saransh719** [comment]
@holocronweaver I am unable to replicate the issue for v2.3.1 of the app on my device, could you send a video of the issue or tell your android version so I can test on an emulator of the same version.

**Comment by iSoron** [comment]
Thanks for the report, @holocronweaver, but this behavior is intentional due to issue https://github.com/iSoron/uhabits/issues/1719. In summary, Samsung's keyboard doesn't allow users to enter decimal values in some locales. See also https://github.com/flutter/flutter/issues/61175.

If this has already been fixed by Samsung, we can remove the workaround.

**Comment by holocronweaver** [comment]
@iSoron the issue I reported is for Swiftkey keyboard, not Samsung. At the time of the report it was a regression that affected all keyboards, but now it seems to just be Swiftkey (and maybe Samsung as well?)

**Comment by iSoron** [comment]
@holocronweaver Sorry, it affected both Swiftkey and Samsung. If anyone wants to comment out that work around and confirm that #1719 is gone, we can remove it.

**Comment by Saransh719** [comment]
> [@holocronweaver](https://github.com/holocronweaver) Sorry, it affected both Swiftkey and Samsung. If anyone wants to comment out that work around and confirm that [#1719](https://github.com/iSoron/uhabits/issues/1719) is gone, we can remove it.

If #1719 is still present. Isn't there any way to check keyboard language and make it so that the full keyboard only appears when keyboard language is set to languages like german where decimal is the point operator?

**Comment by iSoron** [comment]
> Isn't there any way to check keyboard language and make it so that the full keyboard only appears when keyboard language is set to languages like german where decimal is the point operator?

Sure, such a pull request would be fine. For safety, we should enable the workaround whenever the decimal separator is anything other than decimal dot, instead of enabling it only for decimal comma.

**Comment by Saransh719** [comment]
> > Isn't there any way to check keyboard language and make it so that the full keyboard only appears when keyboard language is set to languages like german where decimal is the point operator?
> 
> Sure, such a pull request would be fine. For safety, we should enable the workaround whenever the decimal separator is anything other than decimal dot, instead of enabling it only for decimal comma.

I would have figured out something like that but for some reason, on my devices, even while using samsung or swift keyboard, the numeric pad is opening. Can you think of any reason for that? I have tried on two devices, one with android 10 and other with 15.
